src/macaron/slsa_analyzer/provenance/expectations/cue/cue_validator.py

```diff
@@ -78,20 +76,36 @@ def validate_expectation(expectation: str | None, prov: JsonType) -> bool:
 
     Raises
     ------
-    CUERuntimeError, CUEExpectationError
+    CUERuntimeError
         If expectation is invalid or unable to validate the expectation by invoking the shared library.
     """
-    if not expectation:
-        raise CUEExpectationError("CUE policies is empty.")
-
-    expectation_buffer = ctypes.create_string_buffer(bytes(expectation, encoding="utf-8"))
-    prov_buffer = ctypes.create_string_buffer(bytes(json.dumps(prov), encoding="utf-8"))
-
-    def _errcheck(result: int, func: Callable, args: tuple) -> int:  # pylint: disable=unused-argument
-        if result == -1:
-            raise CUERuntimeError("Unable to validate the CUE expectation")
-        return result
-
-    cue.target.errcheck = _errcheck  # type: ignore
-    result = bool(cue.validate(expectation_buffer, prov_buffer))
-    return result
+    cmd = [
+        os.path.join(MACARON_PATH, "bin", "cuevalidator"),
+        "-validate-policy",
+        expectation_path,
+        "-validate-provenance",
+        prov_stmt_path,
+    ]
+
+    try:
+        result = subprocess.run(  # nosec B603
+            cmd,
+            capture_output=True,
+            check=True,
+            cwd=MACARON_PATH,
+            timeout=defaults.getint("cue_validator", "timeout", fallback=30),
+        )
+    except (
+        subprocess.CalledProcessError,
+        subprocess.TimeoutExpired,
+        FileNotFoundError,
+    ) as error:
+        raise CUERuntimeError("Unable to process CUE expectation or provenance.") from error
+
+    if result.returncode == 0:
```

Comment:
From the doc string in `cmd/cuevalidator/cuevalidator.go`, the return code can indicates whether the validation passed or failed. I wonder if we could rely on that ?

Reply:
The return code will be `0` if no error happens. The value logged in the stdout, i.e., `True` and `False` will report the validation result.